phoenixRest/views/user/instance/__init__.py

```diff
@@ -141,6 +147,23 @@ def get_user(context, request):
         return map_user_with_secret_fields(context.userInstance, request)
     return map_user_public_with_positions(context.userInstance, request)
 
+@view_config(context=UserInstanceResource, name='friendships', request_method='GET', renderer='json', permission='get_friendship_states')
+def get_friendships(context, request):
+    query = request.db.query(Friendship).filter(
+        and_(
+            and_(Friendship.accepted is not None, Friendship.revoked is None),
+            or_(Friendship.recipient_user == context.userInstance, Friendship.source_user == context.userInstance))
+    ).all()
+    return query
+    
+@view_config(context=UserInstanceResource, name='friend_requests', request_method='GET', renderer='json', permission='get_friendship_states')
+def get_friend_requests(context, request):
```

Comment:
We probably want a function to show revoked/rejected requests as well. So the question is:

Do we make a third function, or use one function to see all requests of all states?

In any regard, users should **not** be able to see requests they sent themselves if they are not accepted, to avoid being able to tell if you were rejected or not.

Reply:
> 

I'd say making a third function would be easiest to implement first.

So source users should not be able to see their sent pending friend requests?

Reply:
@simsine Correct. We could also consider not showing the rejected state to the source user, but I think that is prone to error. What do you think is the best solution to this from the perspective of protecting everyone involved?

Reply:
Honestly, I don't really understand what what the issue is here. So you're gonna have to describe how i should implement it.